woocommerce/woocommerce-ios
# REST API: Password login UI

Part of #8918 


## Description

This PR continues the custom WPCom login flow of the Jetpack setup feature by adding the UI for the password screen. Besides the UI part, some logic is also included:
- Constructing Gravatar URL based on the input email to show in the avatar view.
- Handling magic link requests and showing the magic link screen/error upon receiving responses.
- Error handling was updated by getting the `localizedDescription` to display on the alert if possible. This helps show more specific messages.
- A new button style `SecondaryLoadingButtonStyle` to show loading indicator on secondary buttons.

Flow reference: [pe5sF9-1c9-p2].

## Testing instructions

Prerequisite: Make sure that you have access to a site with WooCommerce and no Jetpack.
- Log out of the app or skip onboarding if needed.
- Tap "Enter your site address" on the prologue screen and proceed with the address of your test site.
- Log in with the credentials of an admin.
- When the login completes, tap the Jetback benefit banner at the bottom of the Dashboard screen.
- Tap Log In on the benefit modal.
- Enter the email address of a WPCom account with a password on the presented WPCom email login screen.
- Tap Install/Connect Jetpack. Notice that the password screen is displayed with the correct avatar.
- Tap Continue using magic link. The magic link screen should be displayed after the link is requested successfully.
- Try again with an A8C email address. Tapping Continue using magic link should fails with a message "You're not allowed to request login link for this account".

## Screenshots

<img src="https://user-images.githubusercontent.com/5533851/222445202-5aeaf7e8-e461-439f-acb9-e3c0abd961d9.png" width=320 />


---
